refactor(orders): ORDERS-7036 refactor orders/invoice.html pickup_details section to better align with existing pickup_address output

matt-evangelidis opened the pull request:
#### What?

This PR changes the `pickup_address` fields accessed and displayed in `orders/invoice.html` to match the fields used in `orders/details.html`. 

This change, along with another PR solves an issue where the pickup address would not display on the order invoice or order details pages, if an order with a pickup consignment had only one pickup consignment (which is technically all of them based on current platform limitations).

This is also being done as part of simplifying the backend code that produces the object for `order.pickup_address`, as there is no reason to use two different output formats for the same data.

#### Requirements

- [ ] CHANGELOG.md entry added (required for code changes only)

#### Tickets / Documentation

- [ORDERS-7036](https://bigcommercecloud.atlassian.net/browse/ORDERS-7036)
- https://github.com/bigcommerce/bigcommerce/pull/63822

#### Screenshots (if appropriate)

<details>
<summary>Invoice Page</summary>

Without UI or backend changes
![image](https://github.com/user-attachments/assets/fdf38707-81a9-48db-80b6-dd47d145f64c)


With backend changes only
![image](https://github.com/user-attachments/assets/42f551df-828a-4d91-8ad0-c47fef5bf610)


With UI and backend changes
![image](https://github.com/user-attachments/assets/48d6e602-a274-404b-a98c-30b8c50be24a)


</details>

donald-nguyen-bc reviewed (approved):
👍

bc-vincent-zhao reviewed (commented):
For context: I think the code look like this is due to the MLI phasing, strictly speaking, we've only delivered the "single pickup" phase, so you would see assumption made throughout the codebase that there's one pickup address to simplify things. 

Some engineers find it more future proof to handle multiple pickup case to start with so we don't have to come back and revisit this, which led to the current inconsistency.

For rolling out this change:

This change depends on a new data structure - a list of pickup address instead of a single one, so for it to work, the new data structure needs to be always there, so given we don't have experiment control in cornerstone, the best way to introduce this change is to make the **additional** data available from bcapp side.

So for the default experience for any new stores or existing store, they will benefit from this change that now will support multiple pickup addresses.

However, for existing stores that might have customised this template, for backward compatible reason, I think we still need to expose the single pickup object format.

So we should really have 2 data available at the same time:
- order.pickup_address -- existing 1st pickup address
- order.pickup_addresses -- new additive change that expose the collection

jkanive commented:
@matt-evangelidis Could you point me to the different in `before` and `after` screenshots? I might be missing something.

bc-alexsaiannyi commented:
Hi @matt-evangelidis, changes themselves look fine. Just small comments:

1.  Please add item to `changelog` regarding your change 💅
2. I like Vincent idea about backward compatibility. Can you expand where we are with this? and do I understand this correctly that new data `order.pickup_addresses` covered with experiment `ORDERS-7036.SF_order_details_page_unify_pickup_address` and if not we back to `order.pickup_address`?

matt-evangelidis commented:
> Hi @matt-evangelidis, changes themselves look fine. Just small comments:
> 
> 1. Please add item to `changelog` regarding your change 💅
> 2. I like Vincent idea about backward compatibility. Can you expand where we are with this? and do I understand this correctly that new data `order.pickup_addresses` covered with experiment `ORDERS-7036.SF_order_details_page_unify_pickup_address` and if not we back to `order.pickup_address`?

Hi @bc-alexsaiannyi, 

I've added a `changelog` entry under `Draft`.

To explain the backwards compatibility concept, the existing payloads that the Order Details and Order Invoice pages get from BCApp contains a `pickup_address` field. Previously, this object would have different keys depending on the [condition evaluated here](https://github.com/bigcommerce/bigcommerce/blob/0b0633112280376ea11eca30a976f634024378bb/includes/resources/AccountOrderDetails.php#L614), so depending on the order, `pickup_address` would have the correct keys for one page but not the other.

To solve this, we've added a net-new `pickup_addresses` field which has no conditional behaviour and will have a guaranteed set of keys that both pages can safely read from (we're no longer covering this with an experiment, given this is completely net-new behaviour with no existing usage). To keep previous versions of Cornerstone working, we'll keep the `pickup_address` field around and with its current behaviour.

matt-evangelidis commented:
Would appreciate any reviews from @bigcommerce/team-store-design

bc-alexsaiannyi commented:
> ... we'll keep the `pickup_address` field around and with its current behaviour.

@matt-evangelidis, thanks for giving more context on this. It sounds good to me.

bc-alexsaiannyi reviewed (approved):
lgtm
